x-pack/plugins/security_solution/public/cases/components/case_view/index.test.tsx

```diff
@@ -723,6 +723,32 @@ describe('CaseView ', () => {
     });
   });
 
+  it('should show the correct connector name on the push button', async () => {
+    useConnectorsMock.mockImplementation(() => ({ connectors: connectorsMock, loading: false }));
+    useGetCaseUserActionsMock.mockImplementation(() => ({
+      ...defaultUseGetCaseUserActions,
+      hasDataToPush: true,
+    }));
+
+    const wrapper = mount(
+      <TestProviders>
+        <Router history={mockHistory}>
+          <CaseComponent {...{ ...caseProps, connector: { ...caseProps, name: 'old-name' } }} />
+        </Router>
+      </TestProviders>
+    );
+
+    await waitFor(() => {
+      expect(
+        wrapper
+          .find('[data-test-subj="has-data-to-push-button"]')
+          .first()
+          .text()
+          .includes('My Connector 2')
```

Comment:
Sorry for the dumb question but how does the new name get set? Is it changed by a different test? Or in one of the mock implementations above?

Reply:
Hey! It is set in `useConnectorsMock.mockImplementation(() => ({ connectors: connectorsMock, loading: false }));`.